### PR TITLE
New Header

### PR DESCRIPTION
**Pull request by srgzlb**
The purpose of this PR is to implement the Header

Requirements:
1. The "Menu" button can be positioned on the left or right side on mobile and only right side on the desktop.

2. There are 2 header background color options: transparent or background-primary. If the transparent option is chosen should be added padding-top to the next section. At the moment there is CSS variable --header-height that will be used for padding top in section below header.

3. If the page is scrolled down and the background color is set to transparent, then the background should be changed to background-primary color as well. If the page is scrolled to top then the header should have a transparent background color.

4. Also, the background color should be changed to background-primary color if the menu or search is opened, no matter what background color option was chosen or what is current scroll position. The initial background color should be applied when the menu or search is closed.

5. The Period picker should be placed below the header on mobile and in the center of header on the desktop. Currently, no extra requirements implemented for it. Were decided to leave it as it is now and return to the development later if needed.

Google Chrome_2023-12-05 15-02-00@2x
Google Chrome_2023-12-05 15-02-24@2x
Google Chrome_2023-12-13 10-39-09@2x

Teampaper_2023-12-05 15-00-44@2x

### CODE REVIEW

File: layout/theme.liquid [hunk 1]
```diff
@@ -17,16 +17,14 @@
...
-      </script>
-    {%- endif -%}
+    <script>
+      // Loading the IFrame Player API code asynchronously.
+      const tag = document.createElement('script');
+      tag.src = "https://www.youtube.com/iframe_api";
+
+      const firstScriptTag = document.getElementsByTagName('script')[0];
+      firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+    </script>
```
Comment (thecoorum) [inline layout/theme.liquid:27]: Won't the `<script src="..." async></script>` work here?

Comment (srgzlb) [reply layout/theme.liquid:27]: Unfortunately, this approach doesn't work but I found a new solution for this issue. I've already implemented and checked it in the Chameleon theme, seems it works properly on all devices I have. 
So I will share this solution with Impact theme but when I will work with Hero section.

Comment (thecoorum) [reply layout/theme.liquid:27]: Okay, then I think you can at least simplify the approach with something like the following
```liquid
<script>
  const tag = document.createElement('script');

  tag.src = 'https://www.youtube.cim/iframe_api';

  document.body.appendChild(tag);
</script>
```

Comment (srgzlb) [reply layout/theme.liquid]: FIxed

**Review by thecoorum — APPROVED** [review]

**Pull request merged**